e2e/interchaintestv8/sp1_ics07_test.go

```diff
@@ -214,3 +233,86 @@ func (s *SP1ICS07TendermintTestSuite) TestUpdateClientAndMembership() {
 		s.Require().False(clientState.IsFrozen)
 	}))
 }
+
+// TestMisbehaviour tests the misbehaviour functionality
+// Partially based on https://github.com/cosmos/relayer/blob/f9aaf3dd0ebfe99fbe98d190a145861d7df93804/interchaintest/misbehaviour_test.go#L38
+func (s *SP1ICS07TendermintTestSuite) TestMisbehaviour() {
+	ctx := context.Background()
+
+	s.SetupSuite(ctx)
+
+	eth, simd := s.ChainA, s.ChainB
+	_ = eth
+
+	var height clienttypes.Height
+	var oldHeader tmclient.Header
+	s.Require().True(s.Run("Get old header", func() {
+		var latestHeight int64
+		var err error
+		oldHeader, latestHeight, err = ibcclientutils.QueryTendermintHeader(simd.Validators[0].CliContext())
+		s.Require().NoError(err)
+		s.Require().NotZero(latestHeight)
+
+		height = clienttypes.NewHeight(clienttypes.ParseChainID(simd.Config().ChainID), uint64(latestHeight))
+
+		clientState, err := s.contract.GetClientState(nil)
+		s.Require().NoError(err)
+		trustedHeight := clienttypes.NewHeight(uint64(clientState.LatestHeight.RevisionNumber), uint64(clientState.LatestHeight.RevisionHeight))
+
+		oldHeader.TrustedHeight = trustedHeight
+		oldHeader.TrustedValidators = oldHeader.ValidatorSet
+	}))
+
+	s.Require().True(s.Run("Invalid misbehaviour", func() {
+		// Create a new valid header
+		newHeader := s.CreateTMClientHeader(
+			ctx,
+			simd,
+			int64(height.RevisionHeight+1),
+			oldHeader.GetTime().Add(time.Minute),
+			oldHeader,
+		)
+
+		invalidMisbehaviour := tmclient.Misbehaviour{
+			Header1: &newHeader,
+			Header2: &oldHeader,
+		}
+
+		// The proof should fail because this is not misbehaviour (valid header for a new block)
+		_, err := operator.Misbehaviour(simd.GetCodec(), invalidMisbehaviour, false,
+			"--trust-level", testvalues.DefaultTrustLevel.String(),
+			"--trusting-period", strconv.Itoa(testvalues.DefaultTrustPeriod))
+		s.Require().ErrorContains(err, "Misbehaviour is not detected")
+	}))
+
+	s.Require().True(s.Run("Valid misbehaviour", func() {
```

Comment:
nit: Can we add a test with time montonicity breaking. ie, next header has a lower time

Reply:
Yes, that makes perfect sense.